src/main/java/org/folio/service/pieces/flows/delete/PieceDeleteFlowInventoryManager.java

```diff
@@ -0,0 +1,61 @@
+package org.folio.service.pieces.flows.delete;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.folio.models.ItemStatus;
+import org.folio.models.pieces.PieceDeletionHolder;
+import org.folio.orders.utils.RequestContextUtil;
+import org.folio.rest.core.models.RequestContext;
+import org.folio.rest.jaxrs.model.Piece;
+import org.folio.service.inventory.InventoryItemManager;
+import org.folio.service.pieces.PieceUpdateInventoryService;
+
+import java.util.Optional;
+
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS;
+import static org.folio.service.inventory.InventoryItemManager.ITEM_STATUS_NAME;
+
+public class PieceDeleteFlowInventoryManager {
+
+  private final InventoryItemManager inventoryItemManager;
+  private final PieceUpdateInventoryService pieceUpdateInventoryService;
+  public PieceDeleteFlowInventoryManager(InventoryItemManager inventoryItemManager,
+                                         PieceUpdateInventoryService pieceUpdateInventoryService) {
+    this.inventoryItemManager = inventoryItemManager;
+    this.pieceUpdateInventoryService = pieceUpdateInventoryService;
+  }
+
+  public Future<Pair<String, String>> processInventory(PieceDeletionHolder holder, RequestContext requestContext) {
+    var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, holder.getPieceToDelete().getReceivingTenantId());
+    return deleteItem(holder, locationContext)
+      .compose(aVoid -> holder.isDeleteHolding()
+          ? pieceUpdateInventoryService.deleteHoldingConnectedToPiece(holder.getPieceToDelete(), locationContext)
+          : Future.succeededFuture()
+      );
+  }
+
+  private Future<Void> deleteItem(PieceDeletionHolder holder, RequestContext requestContext) {
+    var piece = holder.getPieceToDelete();
+    return getOnOrderItemForPiece(piece, requestContext)
+      .compose(item -> Optional.ofNullable(item)
+        .map(mItem -> inventoryItemManager.deleteItem(piece.getItemId(), true, requestContext))
+        .orElse(Future.succeededFuture()));
+  }
+
+  private Future<JsonObject> getOnOrderItemForPiece(Piece piece, RequestContext requestContext) {
+    return StringUtils.isEmpty(piece.getItemId())
+      ? Future.succeededFuture()
+      : inventoryItemManager.getItemRecordById(piece.getItemId(), true, requestContext)
+          .map(item -> getItemWithStatus(item, ItemStatus.ON_ORDER.value()));
```

Comment:
ternary operators is good to use but sometimes, make it less readiable long chain of codes. we can do it like this
```
if (StringUtils.isEmpty(piece.getItemId())) {
      return Future.succeededFuture()
}
return inventoryItemManager.getItemRecordById(piece.getItemId(), true, requestContext)
    .map(item -> getItemWithStatus(item, ItemStatus.ON_ORDER.value()));
```

Reply:
I think ternary is okay with 1 or 2 lines of code but if it was less readable for you, surely I will change this